src/QMCWaveFunctions/RotatedSPOs.cpp

```diff
@@ -172,6 +191,41 @@ void RotatedSPOs::apply_rotation(const std::vector<RealType>& param, bool use_st
   Phi->applyRotation(rot_mat, use_stored_copy);
 }
 
+void RotatedSPOs::constructDeltaRotation(const std::vector<RealType>& delta_param,
+                                         const std::vector<RealType>& old_param,
+                                         const RotationIndices& act_rot_inds,
+                                         const RotationIndices& full_rot_inds,
+                                         std::vector<RealType>& new_param,
+                                         ValueMatrix& new_rot_mat)
+{
+  assert(delta_param.size() == act_rot_inds.size());
+  assert(old_param.size() == full_rot_inds.size());
+  assert(new_param.size() == full_rot_inds.size());
+
+  const size_t nmo = new_rot_mat.rows();
+  assert(new_rot_mat.rows() == new_rot_mat.cols());
+
+  ValueMatrix old_rot_mat(nmo, nmo);
+  old_rot_mat = ValueType(0);
+
+  constructAntiSymmetricMatrix(full_rot_inds, old_param, old_rot_mat);
```

Comment:
Is the set 0 necessary? constructAntiSymmetricMatrix doesn't sound like need of setting zero before calling it.

Reply:
constructAntiSymmetric matrix does need the input matrix to be zero.  Not all entries are referenced, and the unreferenced entries will not get set by the function.
It looks like ValueMatrix uses Vector as an underlying container and that will set the value to zero on initialization/resize.

Reply:
`unreferenced entries will not get set by the function.` that means `constructAntiSymmetric` assumes the previous value should be kept untouched. Is a desired safe behavior? I'm wondering if constructAntiSymmetric should set 0 internally. Whether ValueMatrix set 0 or not at initialization is not relevant when we define the behavior of constructAntiSymmetric I assume.

Reply:
There's a comment in constructAntiSymmetricMatrix that states the assumption that the rotation matrix is initialized to zero upon entry.   Based on your other review comments on this PR, I got the impression you were concerned about performance implications, and in this case it is redundantly setting the matrix to zero.    But constructAntiSymmetricMatrix could set the array to zero to ensure correctness.

Reply:
In this case, correctness is favored.